plugins/modules/win_firewall.ps1

```diff
@@ -13,11 +13,15 @@ $check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "b
 
 $profiles = Get-AnsibleParam -obj $params -name "profiles" -type "list" -default @("Domain", "Private", "Public")
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -failifempty $true -validateset 'disabled','enabled'
+$inbound_action = Get-AnsibleParam -obj $params -name "inbound_action" -type "str" -default "block" -validateset 'allow','block'
+$outbound_action = Get-AnsibleParam -obj $params -name "outbound_action" -type "str" -default "allow" -validateset 'allow','block'
 
 $result = @{
     changed = $false
     profiles = $profiles
     state = $state
+    inbound_action = $inbound_action
```

Comment:
While you can keep it, I personally try to avoid returning the same data that the user already knows. If the module is converted to the newer Ansible.Basic.AnsibleModule wrapper than module invocation data is added by default. Return values should be reserved for data that is generated by the module or information the user may not know already. Anything else is mostly just historical behaviour before we tried to make that policy.

Reply:
Yes, but the profile and state are also returning which is passed by the user and the user already knows about it, which is why I returned inbound/outbound actions.  Thanks for the explanation though. Really helpful